### PR TITLE
Reworked params into wrapper function

### LINKED ISSUE #38: Proper way of creating scalar with parameters
First of all, thanks for creating this awesome library. My question is, how should I create a field that takes parameters, but returns a scalar instead of a nested object?

Example:
```gql
type Query {
  getContract: Contract
}

type Contract {
   title: String!
   isValid(timestamp: String!): Boolean!
}
```
Desired query:
```gql
query {
  getContract {
    title: String!
    isValid(timestamp: "2019-01-30")
  }
}
```
I can't use __params, because graphqlify will generate an empty { } block after the name and parameters.

My workaround:
```ts
const contractObj = {
  title: types.string,
  isValid: types.boolean, // for type inference and auto-compĺetion to work properly
}

delete contractObj.isValid;
contractObj['isValid(timestamp: "2019-01-30")'] = types.boolean;

const queryObj = { getContract: contractObj };
const queryString = graphqlify.query('MyQuery', queryObj);
```
I could have added `['isValid(timestamp: "2019-01-30")']: types.boolean` to the object literal, but then auto-completion, type inference, etc, wouldn't work as desired.

Is something like this possible?
```ts
const contractObj = {
  title: types.string,
  isValid: withParameters(types.boolean, { timestamp: '"2018-01-01"' }),
};
```

### PR DESCRIPTION
**Pull request by luvies**
This PR will close #38 and implements the `params` function to add parameters to a given GQL query. This is a breaking change.

Old system:
```ts
const queryObject = {
  user: {
    __params: { id: 1 },
    id: types.number,
    name: types.string,
    bankAccount: {
      id: types.number,
      branch: types.string,
    },
  },
}
```

New system:
```ts
const queryObject = {
  user: params({ id: 1 }, {
    id: types.number,
    name: types.string,
    bankAccount: {
      id: types.number,
      branch: types.string,
    },
  }),
}
```

I've gone with `params` as the function name since it's better to prefer the explicit name over the regularly used `$`, and let users just do

```ts
import { params as $ } from 'typed-graphqlify';
```

where it makes sense for them.

Notes: Before releasing a new version with this, we need to sort 2 other things first:

- I will have a PR adding fragment support very shortly (so it would be worth releasing the changes together)
- Since this is a breaking change, it might be worth doing the function export change that was mentioned as well, so both breaking changes are done at once

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,11 +2,11 @@
 [![npm version](https://badge.fury.io/js/typed-graphqlify.svg)](https://badge.fury.io/js/typed-graphqlify)
```
Comment (acro5piano) [inline README.md]: Thank you for fixing a lot of wrong grammars...

File: src/graphqlify.ts [hunk 2]
```diff
@@ -25,6 +25,18 @@ export const graphqlify = {
   subscription: createOperate('subscription'),
 }
```
Comment (acro5piano) [inline src/graphqlify.ts]: Your refactoring made it easy to add new features 🎉

**Comment by luvies** [comment]
I opened up the fragments PR, so we should be good on this?

Do we want to adjust the exports so that `query` and `mutation` are exported directly? If so, we should do it before making a new release. It should be a small change, but it will need to be done after these 2 PRs are merged, since we will have to adjust the tests as well.

**Comment by acro5piano** [comment]
Thank you for great contribution!

> Do we want to adjust the exports so that `query` and `mutation` are exported directly?

Yes, direct export is simpler as we talked before. I will do it just after merging your PRs!

**Pull request merged**